### PR TITLE
Fixed compilation warnings

### PR DESCRIPTION
**Pull request by radarhere**
Attempt to fix several `comparison between signed and unsigned integer expressions` warnings generated on Travis.

### CODE REVIEW

**Comment by wiredfool** [comment]
* This has some conflicts with #1663 
* We should be more careful about casting, and not do it without range checks when the actual range of the item is unknown or controlled by a file format.

**Comment by radarhere** [comment]
In my attempt to rebase, I've accidentally removed the first commit. So now this PR just changes several conversions from implicit to explicit.

**Comment by aclark4life** [comment]
What's going on here? Any labels that can be applied to make it easier to ascertain the status? Thanks

**Comment by radarhere** [comment]
'Needs Review' is fair. This is now just trying to fix some compilation warnings by changing several conversions from implicit to explicit.

File: src/_imaging.c [hunk 1]
```diff
@@ -406,11 +406,11 @@ getlist(PyObject* arg, Py_ssize_t* length, const char* wrong_length, int type)
         // on this switch. And 3 fewer loops to copy/paste.
         switch (type) {
         case TYPE_UINT8:
-            itemp = PyInt_AsLong(op);
+            itemp = (int)PyInt_AsLong(op);
```
Comment (homm) [inline src/_imaging.c]: Maybe it's better to declare itemp as long?
https://docs.python.org/3/c-api/long.html#c.PyLong_AsLong

File: src/_imaging.c [hunk 2]
```diff
@@ -506,7 +506,7 @@ getink(PyObject* color, Imaging im, char* ink)
 #else
                 if (PyInt_Check(color) || PyLong_Check(color)) {
                         if (PyInt_Check(color))
-                                r = PyInt_AS_LONG(color);
+                                r = (int)PyInt_AS_LONG(color);
```
Comment (homm) [inline src/_imaging.c]: The comment near r declaration mentions that r could be up to 0xFFFFFFFF, which is overflow for `int`.
r is declared as PY_LONG_LONG, PyInt_AS_LONG really returns `PyIntObjectob_ival.ob_ival`, which is declared as long.

Is there really was a warning in this place?

Comment (radarhere) [reply src/_imaging.c]: I actually can no longer find these warnings on Travis. So, feel free to pick up this PR if you want, but for my part, without a problem to solve, I am willing to consider this resolved.

File: src/_imaging.c (hunk 1 above)
Comment (radarhere) [reply src/_imaging.c]: Thanks for your thoughts. See my previous comment about this being resolved.

**Pull request closed without merging**